### PR TITLE
Add scale property to control cross fade icon animation time

### LINKED ISSUE #1064: Cross fade icon animation time
## Describe the feature
The cross fade icon animation is now hardcoded to value 200, it would be awesome if this could be changed through props.

https://github.com/callstack/react-native-paper/blob/f4ef4bddc28764d8f87b3f98ccf8773d902535d2/src/components/CrossFadeIcon.js#L60-L64

## Motivation
I have a collapse functionality in my application, upon collapsing it executes a animation, however this animation is 300ms instead of 200ms, so the cross fade icon animation finishes first which I do not want.

## Related Issues
None I think

### PR DESCRIPTION
**Pull request by Taym95**
Try to fix #1064

### CODE REVIEW

File: src/components/CrossFadeIcon.js [hunk 1]
```diff
@@ -47,6 +53,11 @@ class CrossFadeIcon extends React.Component<Props, State> {
 
   componentDidUpdate(prevProps: Props, prevState: State) {
     const { previousIcon } = this.state;
+    const {
+      theme: {
+        animation: { scale },
```
Comment (brunohkbx) [inline src/components/CrossFadeIcon.js]: Shouldn't it have a default value in [DefaultTheme](https://github.com/callstack/react-native-paper/blob/master/src/styles/DefaultTheme.js) ?

Comment (Taym95) [reply src/components/CrossFadeIcon.js]: I think [withTheme](https://github.com/callstack/react-native-paper/blob/master/src/core/theming.js) uses [DefaultTheme](https://github.com/callstack/react-native-paper/blob/master/src/styles/DefaultTheme.js)

Comment (brunohkbx) [reply src/components/CrossFadeIcon.js]: I meant the value for `scale`. Clicking on the FAB button in the example app have thrown an exception when I tried your changes without defining it in DefaultTheme:

```
animation: {
  scale: 1,
}
```

WhatsApp Image 2019-05-15 at 14 28 29

Comment (Taym95) [reply src/components/CrossFadeIcon.js]: Does exist in branch 3.0 [DefaultTheme](https://github.com/callstack/react-native-paper/blob/3.0/src/styles/DefaultTheme.js)

**Comment by ferrannp** [comment]
Thank you!

**Pull request merged**